var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -53,14 +53,15 @@ class Cp2k(MakefilePackage, CudaPackage):
                          ' and BQB compression'))
     variant('spglib', default=False, description='Enable support for spglib')
 
-    variant('cuda_arch_35_k20x', default=False,
-            description=('CP2K (resp. DBCSR) has specific parameter sets for'
-                         ' different GPU models. Enable this when building'
-                         ' with cuda_arch=35 for a K20x instead of a K40'))
-    variant('cuda_fft', default=False,
-            description=('Use CUDA also for FFTs in the PW part of CP2K'))
-    variant('cuda_blas', default=False,
-            description=('Use CUBLAS for general matrix operations in DBCSR'))
+    with when('+cuda'):
+        variant('cuda_arch_35_k20x', default=False,
+                description=('CP2K (resp. DBCSR) has specific parameter sets for'
+                             ' different GPU models. Enable this when building'
+                             ' with cuda_arch=35 for a K20x instead of a K40'))
```

Comment:
Not relevant to your changes, but damn this is ugly 😂

Reply:
Well, luckily Nvidia fixed that afterwards, and many codes can build for multiple models in parallel, hence don't have that problem in the first place.